### PR TITLE
Implement a page view id.

### LINKED ISSUE #969: Pass page view id and a client identifier to ads, analytics, etc.
We need a general variable available to amp-pixel, analytics, ads, that is "unique" (Just Math.random should be fine) per page view. So, just a random number we create once per page.

This is a change in
https://github.com/ampproject/amphtml/blob/master/src/url-replacements.js

Some ads require access to a CID. Create configuration per ad network for the fallbackCookieName. Will need to figure out how to get consent in this case.

CC @rudygalfi @kashyapnitin @avimehta

### PR DESCRIPTION
**Pull request by cramforce**
The id is relatively likely to be unique per page load, URL and user.

Fixes #969

### CODE REVIEW

**Comment by cramforce** [comment]
Fixed test

File: src/url-replacements.js [hunk 1]
```diff
@@ -14,15 +14,16 @@
...
+import {documentInfoFor} from './document-info';
+import {getService} from './service';
+import {parseUrl, removeFragment} from './url';
 
 
 /**
  * This class replaces substitution variables with their values.
  */
-export class UrlReplacements {
+class UrlReplacements {
```
Comment (dvoytenko) [inline src/url-replacements.js]: These were two reasons why I exported the class:
1. Avi was saying that he wanted to customize url replacements for analytics that only applied there.
2. I wanted to keep it extensible for amp-list for infinite scroll. E.g. "https://service/list.json?page=PAGE", where a hypothetical "PAGE" var would only be available in the context of amp-list.

Comment (cramforce) [inline src/url-replacements.js]: I think we can do it when we get there. For now a service seems a natural fit here. I think for other use cases, just replacing the raw replacement functionality might be better than inheritance.

Comment (dvoytenko) [inline src/url-replacements.js]: It's been intended as composition, not inheritance. If we proceed as a singleton, let's hide "set" method.

**Comment by dvoytenko** [comment]
LGTM once `set` method is removed/hidden.

**Pull request merged**